src/network/clientpackethandler.cpp

```diff
@@ -994,14 +994,22 @@ void Client::handleCommand_AddParticleSpawner(NetworkPacket* pkt)
 	p.amount             = readU16(is);
 	p.time               = readF32(is);
 
-	// older protocols do not support tweening, and send only
-	// static ranges, so we can't just use the normal serialization
-	// functions for the older values.
-	p.pos.start.legacyDeSerialize(is);
-	p.vel.start.legacyDeSerialize(is);
-	p.acc.start.legacyDeSerialize(is);
-	p.exptime.start.legacyDeSerialize(is);
-	p.size.start.legacyDeSerialize(is);
+	bool missing_end_values = false;
+	if (m_proto_ver >= 42) {
+		// All tweenable parameters
+		p.pos.deSerialize(is);
+		p.vel.deSerialize(is);
+		p.acc.deSerialize(is);
+		p.exptime.deSerialize(is);
+		p.size.deSerialize(is);
```

Comment:
I guess, the conciseness of this code is one of the goals of this PR, but if we are to add a serializable field to TweenableParameter at a later point, will this code survive it?

Reply:
Yes, although the code will be cluttered with passing the protocol version number from one function to the other. I also considered that but resorted to this simpler (but similar future-proof) approach to avoid changing even more lines than necessary.